pkg/reconciler/labeler/accessors.go

```diff
@@ -101,3 +133,11 @@ func (c *configuration) patch(ns, name string, pt types.PatchType, p []byte) err
 	_, err := c.r.client.ServingV1().Configurations(ns).Patch(name, pt, p)
 	return err
 }
+
+func (c *configuration) makeMetadataPatch(ns, name string, routeName *string) (map[string]interface{}, error) {
+	config, err := c.r.configurationLister.Configurations(ns).Get(name)
+	if err != nil {
+		return nil, err
+	}
+	return makeMetadataPatch(config, routeName)
```

Comment:
can't a configuration be used by more than one route?

Reply:
actually no! The existing design puts route names in labels and each serving rev/config can only be referenced once! I'd love to address this by instead making route an annotation with a comma separated list or removing it altogether, but that will have to wait for the future.

Reply:
hm, I thought when I create individual routes and configurations, i.e. not owned by service we can do many to many... but may be not :)

Reply:
I think if you, for example, pinned the same revision to two routes the labeler would start failing here:
https://github.com/knative/serving/blob/81b56954a02e978015437217e6a37e93f731e70b/pkg/reconciler/labeler/labels.go#L80

The losing route would bail out here and also stop labeling Configurations on the next lines (it would fail here the same way every time) and you'd start to see new deployments never become routable. That's my theory at least. I'm not a big fan of the design, but I also don't think there are many real users / tests of those scenarios. It's actually weird that we have this labeler rather than having this occur during the Route reconciler. We already retrieve all Revisions and you can diff the old Status to know which revisions to mark inactive.